packetbeat/beat/packetbeat.go

```diff
@@ -205,8 +197,63 @@ func (pb *Packetbeat) Setup(b *beat.Beat) error {
 	return err
 }
 
+func (pb *Packetbeat) setupSniffer(pub publisher.Client) error {
+	cfg := &pb.PbConfig
+
+	withVlans := cfg.Interfaces.With_vlans
+	withICMP := cfg.Protocols.Icmp.Enabled
+	defaultBpf := cfg.Interfaces.Bpf_filter
+	if defaultBpf == "" && cfg.Flows == nil {
+		defaultBpf = protos.Protos.BpfFilter(withVlans, withICMP)
+	}
+
+	pb.Sniff = &sniffer.SnifferSetup{}
+	return pb.Sniff.Init(false,
+		func(dl layers.LinkType) (sniffer.Worker, string, error) {
```

Comment:
I think it would probably make it more readable to put this in an independent function as it is quite large.